src/main.rs

```diff
@@ -352,11 +353,7 @@ fn fetch_master_commit_via_http(
         status => bail!("received status {} for URL {}", status, URL),
     }
     let master_commit = response.text()?;
-    if master_commit.len() == 40
-        && master_commit
-            .chars()
-            .all(|c| '0' <= c && c <= '9' || 'a' <= c && c <= 'f')
-    {
+    if master_commit.len() == 40 && master_commit.chars().all(|c| c.is_ascii_hexdigit()) {
```

Comment:
I could see, that you would like to reject upper case chars, in that case I would suggest

```
('0'..='9').contains(c) && ('a'..='f').contains(c)
```

Reply:
done. Nice solution, haven't thought about `matches!` in this case.
Maybe it's worth to tell the clippy people ;)